src/main/kotlin/no/nav/medlemskap/regler/v1/ReglerForLovvalg.kt

```diff
@@ -80,13 +182,180 @@ class ReglerForLovvalg(
+            beskrivelse = "",
+            ytelse = ytelse,
+            operasjon = {sjekkOmBrukersBarnErBosattINorge()}
+    )
+
+    private val harBrukerJobbet100ProsentEllerMer= Regel(
+            identifikator = "",
+            avklaring = "",
+            beskrivelse = "",
+            ytelse = ytelse,
+            operasjon = {sjekkOmBrukersStillingsprosentErMerEnn(100.0)}
+    )
+
     private val harBrukerJobbet25ProsentEllerMer = Regel(
             identifikator = "12",
             avklaring = "Har bruker vært i minst 25% stilling de siste 12 mnd?",
             beskrivelse = "",
             ytelse = ytelse,
-            operasjon = { sjekkOmBrukerHarJobbet25ProsentEllerMer() }
+            operasjon = { sjekkOmBrukersStillingsprosentErMerEnn(25.0) }
+    )
+
+    /*
+    private val harBrukerNorsdiskStatsborgerskap = Regel (
+            identifikator = "",
+            avklaring = "Har bruker nordisk statsborgerskap?",
+            beskrivelse = "",
+            ytelse = ytelse,
+            operasjon = {sjekkOmBrukerHarNordiskStatsborgerskap() }
     )
+    */
+
+
+    private val harBrukerBarnUtenEktefelle = Regel (
+            identifikator = "11.2.1",
+            avklaring = "Har bruker uten ektefelle barn registrert i TPS?",
+            beskrivelse = "",
+            ytelse = ytelse,
+            operasjon = {sjekkOmBrukerHarBarn()}
+    )
+
+    private val erBarnløsBrukersEktefelleBosattINorge = Regel (
+            identifikator = "11.3.1",
+            avklaring = "Er brukers ektefelle folkeregistrert som bosatt i Norge?",
+            beskrivelse = "",
+            ytelse = ytelse,
+            operasjon = {sjekkOmBrukersEktefelleErBosattINorge()}
+    )
+
+    private fun sjekkOmBrukerHarBarn(): Resultat {
+        return when {
+            barnITps.erIkkeTom() -> ja()
+            else -> nei("Bruker har ikke barn i tps")
+        }
+     }
+
+    private fun sjekkOmBrukerHarEktefelle(): Resultat {
+      return when {
+        ektefellerITps.erIkkeTom() -> ja()
+        else -> nei("Bruker har ikke ektefelle i tps")
+     }
+    }
+
+    private fun sjekkOmBrukersEktefelleErBosattINorge(): Resultat {
+        val bostedsadresserTilEktefelle = ektefellerITps.flatMap { it.bostedsadresser.adresserSiste12Mnd(kontrollPeriodeForPersonhistorikk) }
+        val postAdresseTilEktefelle = ektefellerITps.flatMap { it.postadresser.landkodeTilAdresseSiste12Mnd(kontrollPeriodeForPersonhistorikk) }
+        val midlertidigPostadresseTilEktefelle = ektefellerITps.flatMap { it.midlertidigAdresser.landkodeTilAdresseSiste12Mnd(kontrollPeriodeForPersonhistorikk) }
+             return when {
+                erPersonBosattINorge(bostedsadresserTilEktefelle, postAdresseTilEktefelle, midlertidigPostadresseTilEktefelle) -> ja()
+                else -> nei("Ikke alle adressene til ektefelle er norske, eller ektefelle mangler bostedsadresse")
+        }
+    }
+
+    private fun sjekkOmBrukersBarnErBosattINorge(): Resultat {
```

Comment:
Synes det ser bra ut, men jeg stusset på prosesstegningen. Hvordan skal pilen med Ja & Nei som går til uavklart tolkes?

Reply:
Ja, var egentlig noe jeg ønsket å prate med henne om at den ble litt vanskelig å løse. Er hvis bruker har flere barn hvor barna har forskjellige adresser, dvs. noen er folkeregistrert og andre ikke. Har foreløpig bare gjort sjekk på om alle barna har norsk adresse eller ikke.

Reply:
Okay, skjønner. Noe sånt som:
barnITps.any { erPersonBosattINorge(it.bostedsadresser, it.postadresser, it.midlertidigAdresser) }
                && barnITps.any { !erPersonBosattINorge(it.bostedsadresser, it.postadresser, it.midlertidigAdresser)}

Ser ut til at det kan være behov for å ha tre utfall på en regel, hvor uavklart blir et nytt mulig utfall, slik vi har sett et behov for og snakket om tidligere. Uavklart blir forøvrig pr def ikke et svar på en regel, men en konklusjon. Tror allikevel det kan være nyttig og vil gjøre at vi får færre regler og dermed enklere regelstruktur. Det blir i såfall en større endring, og passer som egen sak på Trello. Lag en sak på det hvis du er enig. 

Inntil videre kan problemet også løses ved å dele opp spørsmålet i to, slik vi har gjort tidligere.

Reply:
Ja, jeg er enig med deg- Men tenker og at i dette tilfelle er det så mange diamanter at det hadde ikke gjort noe med flere, men godt mulig det på sikt vil være mer løselig. Tenker og kanskje vi kan ta den praten etter Mirna har gjennomgått modellering med Helle. Men kan lage en sak på trello.

Reply:
Dette er løst ved at regelen nå returnerer UAVKLART, og flyten er satt opp til å returnere konklusjon UAVKLART i dette tilfellet. Dette nevnte jeg også på standup på mandag.